src/lmptype.h

```diff
@@ -47,6 +47,8 @@
 #include <cstdint>      // IWYU pragma: export
 #include <cstdlib>      // IWYU pragma: export
 
+#include "library.h"
```

Comment:
Including library.h into lmptype.h is not a good idea. lmptype.h is a "fundamental" header and its contents should be included before others.

Similarly, I would prefer that library.h does not include any LAMMPS headers, so that the library interface is self-contained. That would kill the possible solution to have a new header that can be used by both. It also needs to be usable from C.

I agree with the idea of using the constants from _LMP_DATATYPE throughout and thus avoiding the conversion. Unfortunately, taking all constraints into account would mean we would have to replicate the enum (I would just copy the entire enum and use the same constant names). But this is a section of code that is already replicated, so one more replication is acceptable in my book, especially since this is not a central feature and not likely to be used much beyond PyLammps and including thermo info into dump styles.

Reply:
Ok. Removed the include and duplicated the 3 constants.